### PR TITLE
Allow array items in TypeScript to be nullable

### PR DESCRIPTION
**Pull request by CirnoV**
similar #19070 #19080 

TypeScript Committee: @TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Review by macjohnny — APPROVED** [review]

**Comment by macjohnny** [comment]
@CirnoV thanks for your contribution! please re-generate the samples

**Comment by CirnoV** [comment]
@macjohnny My bad, 😅 I'll commit them shortly

File: samples/client/petstore/typescript-fetch/builds/sagas-and-records/src/apis/PetApiSagas.ts [hunk 1]
```diff
@@ -512,11 +512,11 @@ export interface PayloadUpdatePetRegions extends PayloadUpdatePetRegionsRequest,
 
 export interface PayloadUpdatePetRegionsRequest {
     petId: string;
-    newRegions: List<List<string | null>>;
+    newRegions: List<List<string | null | null>>;
```
Comment (CirnoV) [inline samples/client/petstore/typescript-fetch/builds/sagas-and-records/src/apis/PetApiSagas.ts:515]: It seems I need to modify `typescript-fetch` generator

**Comment by CirnoV** [comment]
I've removed duplicate implemention in `typescript-fetch` generator. It seems to be working now!

**Review by macjohnny — APPROVED** [review]
LGTM

**Comment by msssk** [comment]
This is TS/JS, not Java, null has very little value in JS and should generally be avoided. All of my optional properties that used to have good type definitions like `propName?: string` are now polluted with `propName?: string | null` which propagates to all usage of the value and inserts the possibility of `null` into cases where there will never be `null`.

What drove this change and is there some way in my OAS I can avoid having `| null` appended to all my non-nullable optional properties?

**Pull request merged**